pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -0,0 +1,350 @@
+package multiarchbuildconfig
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/util/retry"
+
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	buildv1client "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
+
+	v1 "github.com/openshift/ci-tools/pkg/api/multiarchbuildconfig/v1"
+	controllerutil "github.com/openshift/ci-tools/pkg/controller/util"
+)
+
+const (
+	controllerName        = "multiarchbuildconfig"
+	nodeArchitectureLabel = "kubernetes.io/arch"
+	noReasonCondition     = "NoReason"
+
+	// TODO: Now we hardcode the URL for the heterogeneous cluster.
+	// TODO: We need to push it to the app.ci registry or quay.io.
+	registryURL = "registry.multi-build01.arm-build.devcluster.openshift.com"
```

Comment:
@hongkailiu Can we start pushing the images to quay.io or we should push them to `app.ci` registry in order to make sure that the test-image-distributor will sync the image across all clusters?

Reply:
Afaik @hongkailiu is syncing images to quay so you would need to treat still app.ci as the source of truth.